src/Microsoft.AspNet.OData.Shared/TypedDelta.cs

```diff
@@ -28,7 +28,7 @@ public abstract class TypedDelta : Delta
         /// <returns>True if it is a Delta generic type; false otherwise.</returns>
         internal static bool IsDeltaOfT(Type type)
         {
-            return type != null && TypeHelper.IsGenericType(type) && type.GetGenericTypeDefinition() == typeof(Delta<>);
+            return type != null && TypeHelper.IsGenericType(type) && (type.GetGenericTypeDefinition() == typeof(Delta<>) || type.GetGenericTypeDefinition() == typeof(DeltaDeletedEntityObject<>) );
```

Comment:
>(type.GetGenericTypeDefinition() == typeof(Delta<>) || type.GetGenericTypeDefinition() == typeof(DeltaDeletedEntityObject<>)  [](start = 69, length = 125)

maybe just use typeof(Delta<>).IsAssignableFrom(type.GetGenericTypeDefinition()) -- that way any type that derives from Delta<T> (with the right T) will be supported, and you don't need to do two (expensive) GetGenericTypeDefinition() reflection calls.